src/dc_mimefactory.rs

```diff
@@ -477,26 +477,27 @@ pub unsafe fn dc_mimefactory_render(mut factory: *mut dc_mimefactory_t) -> libc:
             references_list,
             0 as *mut libc::c_char,
         );
+
+        let os_name = &(*factory).context.os_name;
+        let os_part = os_name
+            .as_ref()
+            .map(|s| format!("/{}", s))
+            .unwrap_or_default();
+        let os_part = to_cstring(os_part);
```

Comment:
Replace this with
```
let os_part = CString::new(os_part).unwrap()
```
and use `os_part.as_ptr()` instead of `os_part` to make Rust manage its memory.

Reply:
Ok.